src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -200,10 +204,14 @@ public function has($id)
             if ('service_container' === $id
                 || isset($this->aliases[$id])
                 || isset($this->services[$id])
-                || array_key_exists($id, $this->services)
             ) {
                 return true;
             }
+            if (array_key_exists($id, $this->services)) {
+                @trigger_error(sprintf('Checking for the existence of a unset "%s" service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
The current behavior is messed up, but untrusty. The property is protected (!). Lets be safe.

If we can go with private.. that be great, as of 4.0 this still can be considered a safeguard sticking when protected.